src/models/PlantModel.ts

```diff
@@ -1,74 +1,83 @@
 
   public async getAll(): Promise<IPlant[]> {
-    const plants = await this.handleFile.readFile<IPlant[]>(FileType.Plants);
+    const query = `SELECT
+      id, breed, size, needs_sun as needsSun, origin, water_Frequency as waterFrequency
+      FROM plants`;
+    const [rows] = await this.conn.execute(query);
+    const plants = rows as IPlant[];
     return plants;
   }
 
   public async create(plant: Omit<IPlant, 'id'>): Promise<IPlant> {
-    const plants = await this.getAll();
-
-    const newPlantId = await this.updateOpsInfo(1);
-    const newPlant = { id: newPlantId, ...plant };
-    plants.push(newPlant);
-
-    await this.handleFile.saveFile(FileType.Plants, plants);
-
+    const {
+      breed, needsSun, origin, size, waterFrequency,
+    } = plant;
+    const query = `INSERT INTO plants (breed, needs_sun, origin, size, water_frequency)
+      VALUES (?, ?, ?, ?, ?)`;
+    const values = [breed, needsSun, origin, size, waterFrequency];
+
+    const [rows] = await this.conn.execute<ResultSetHeader>(query, values);
+
+    const newPlant = {
+      id: rows.insertId,
+      ...plant,
+    };
     return newPlant;
   }
 
   public async getById(id: string): Promise<IPlant | null> {
-    const plants = await this.getAll();
+    const query = `SELECT
+      id, breed, size, needs_sun as needsSun, origin, water_Frequency as waterFrequency
+      FROM plants WHERE id = ?`;
+    const values = [id];
+    const [rows] = await this.conn.execute(query, values);
+
+    const plantById = rows as IPlant[];
+    if (plantById.length === 0) return null;
 
-    const plantById = plants.find((plant) => plant.id === parseInt(id, 10));
-    if (!plantById) return null;
-    return plantById;
+    return plantById[0];
   }
 
   public async removeById(id: string): Promise<IPlant | null> {
-    const plants = await this.getAll();
-
-    const removedPlant = plants.find((plant) => plant.id === parseInt(id, 10));
+    const removedPlant = await this.getById(id);
     if (!removedPlant) return null;
 
-    const newPlants = plants.filter((plant) => plant.id !== parseInt(id, 10));
-    this.handleFile.saveFile(FileType.Plants, newPlants);
+    const query = 'DELETE FROM plants WHERE id = ?';
+    const values = [id];
+    await connection.execute(query, values);
 
     return removedPlant;
   }
 
   public async update(plant: IPlant): Promise<IPlant> {
-    const plants = await this.getAll();
+    const {
+      id, breed, needsSun, origin, size, waterFrequency,
+    } = plant;
 
-    const updatedPlants = plants.map((editPlant) => {
-      if (plant.id === editPlant.id) return { ...plant };
-      return editPlant;
-    });
-
-    await this.handleFile.saveFile(FileType.Plants, updatedPlants);
+    const query = `UPDATE plants 
+      SET breed = ?, needs_sun = ?, origin = ?, size = ?, water_frequency = ?
+      WHERE id = ?`;
+    const values = [breed, needsSun, origin, size, waterFrequency, id];
+    await this.conn.execute(query, values);
```

Comment:
Colocando só pelo reforço mesmo

```suggestion
    await this.conn.execute<OkPacket>(query, values);
```

Reply:
Alteração implementada